src/indent.ts

```diff
@@ -8,6 +8,38 @@ export function newlineAndIndent(
     edit: vscode.TextEditorEdit,
     args: any[]
 ) {
+    // Get rid of any user selected text, since a selection is
+    // always deleted whenever ENTER is pressed.
+    // This should always happen first
+    let selectedText = textEditor.document.getText(textEditor.selection);
+    if (selectedText) {
```

Comment:
Can we use `textEditor.selection.isEmpty`? (See https://code.visualstudio.com/api/references/vscode-api#Selection for all methods/attributes on the selection object.)

Reply:
Yes! I've tested and will update my code.